src/lib/containers/TotalQueryResults.tsx

```diff
@@ -188,47 +199,89 @@ const TotalQueryResults: FunctionComponent<TotalQueryResultsProps> = ({
 
-  const removeFacetSelection = ({
-    facet,
-    selectedValue,
-  }: FacetWithSelection) => {
+  const removeFacetSelection = (facetWithSelection: FacetWithSelection) => {
+    const { facet, selectedValue } = facetWithSelection
     if (facet.facetType === 'enumeration') {
       applyChangesToValuesColumn(
         lastQueryRequest,
         facet as FacetColumnResultValues,
-        applyChanges!,
+        applyChanges,
         selectedValue?.value,
         false,
       )
     } else {
       applyChangesToRangeColumn(
         lastQueryRequest,
         facet as FacetColumnResultRange,
-        applyChanges!,
+        applyChanges,
         [RadioValuesEnum.ANY, RadioValuesEnum.ANY],
       )
     }
   }
 
-  const removeSearchQuerySelection = (columnName: string, value: string) => {
+  const removeColumnSearchQuerySelection = (
+    columnName: string,
+    value: string,
+  ) => {
     const cloneLastQueryRequest = cloneDeep(lastQueryRequest)
     if (!cloneLastQueryRequest.query.additionalFilters) {
       return
     }
-    cloneLastQueryRequest.query.additionalFilters = cloneLastQueryRequest.query.additionalFilters
-      ?.map(el => {
-        return {
-          columnName: el.columnName,
-          values:
-            el.columnName === columnName
-              ? el.values.filter(el => el !== value)
-              : el.values,
-          operator: el.operator,
-          function: el.function,
-          concreteType: el.concreteType
+    cloneLastQueryRequest.query.additionalFilters =
+      cloneLastQueryRequest.query.additionalFilters?.reduce(function (
+        filtered: QueryFilter[],
+        el: QueryFilter,
+      ) {
+        if (
+          isColumnSingleValueQueryFilter(el) ||
+          isColumnMultiValueFunctionQueryFilter(el)
+        ) {
+          // For column-specific QueryFilters, filter out matching values on the matching column
+          const newElement = {
+            columnName: el.columnName,
+            values:
+              el.columnName === columnName
+                ? el.values.filter(el => el !== value)
+                : el.values,
+            operator: isColumnSingleValueQueryFilter(el)
+              ? el.operator
+              : undefined,
+            function: isColumnMultiValueFunctionQueryFilter(el)
+              ? el.function
+              : undefined,
+            concreteType: el.concreteType,
+          }
+
+          // Drop this QueryFilter if it has no more values
+          if (newElement.values.length > 0) {
+            filtered.push(newElement)
+          }
+        } else {
+          // Don't alter/drop non-column-specific QueryFilters
+          filtered.push(el)
         }
-      })
-      .filter(el => el.values.length > 0)
+        return filtered
+      },
+      [])
+
+    executeQueryRequest!(cloneLastQueryRequest)
+  }
+
+  const removeTextMatchesQueryFilter = (
+    queryFilter: TextMatchesQueryFilter,
+  ) => {
```

Comment:
Since the other method only handles Column-specific QueryFilters, we need a new method to handle the new TextMatchesQueryFilter